cpp/src/parquet/column_writer_test.cc

```diff
@@ -1481,7 +1536,7 @@ TEST(TestColumnWriter, WriteDataPagesChangeOnRecordBoundariesWithSmallBatches) {
   auto row_group_reader = file_reader->RowGroup(0);
 
   // Check if pages are changed on record boundaries.
-  const std::array<int64_t, num_cols> expect_num_pages_by_col = {5, 201, 397, 201};
+  const std::array<int64_t, num_cols> expect_num_pages_by_col = {5, 201, 397, 400};
```

Comment:
This column has 150 levels per row and we write 400 rows, 100 levels at a time. The page size is set to 1 byte, so now we get one page written for every  logical row.